### PR TITLE
[CALCITE-6241] Enable a few existing functions to Spark library

### PR DESCRIPTION
**Pull request by YiwenWu**
Add Spark as a supported library for functions that have already been implemented for other libraries.

https://issues.apache.org/jira/browse/CALCITE-6241

### CODE REVIEW

File: site/_docs/reference.md [hunk 1]
```diff
@@ -2800,30 +2805,37 @@ In the following:
...
 | b | PARSE_DATE(format, string)                     | Uses format specified by *format* to convert *string* representation of date to a DATE value
 | b | PARSE_DATETIME(format, string)                 | Uses format specified by *format* to convert *string* representation of datetime to a TIMESTAMP value
 | b | PARSE_TIME(format, string)                     | Uses format specified by *format* to convert *string* representation of time to a TIME value
 | b | PARSE_TIMESTAMP(format, string[, timeZone])    | Uses format specified by *format* to convert *string* representation of timestamp to a TIMESTAMP WITH LOCAL TIME ZONE value in *timeZone*
 | h s | PARSE_URL(urlString, partToExtract [, keyToExtract] ) | Returns the specified *partToExtract* from the *urlString*. Valid values for *partToExtract* include HOST, PATH, QUERY, REF, PROTOCOL, AUTHORITY, FILE, and USERINFO. *keyToExtract* specifies which query to extract
-| b | POW(numeric1, numeric2)                        | Returns *numeric1* raised to the power *numeric2*
+| b s | POW(numeric1, numeric2)                      | Returns *numeric1* raised to the power
+*numeric2*
```
Comment (macroguo-ghy) [inline site/_docs/reference.md:2821]: why use a newline?

**Review by macroguo-ghy — COMMENTED** [review]
Just a question, how to ensure that the semantics of these functions are consistent with those in SPARK?

File: site/_docs/reference.md (hunk 1 above)
Comment (YiwenWu) [reply site/_docs/reference.md]: done, IDEA automatically wraps lines

**Comment by YiwenWu** [comment]
> Just a question, how to ensure that the semantics of these functions are consistent with those in SPARK?

@macroguo-ghy I executed the Function Sqls in the spark engine. And checked the function Definition/OperandType/ReturnType is consistent with the Spark definition.

**Comment by mihaibudiu** [comment]
> > Just a question, how to ensure that the semantics of these functions are consistent with those in SPARK?
> 
> @macroguo-ghy I executed the Function Sqls in the spark engine. And checked the function Definition/OperandType/ReturnType is consistent with the Spark definition.

Some functions have subtle differences for corner cases of the arguments. And Calcite tests often do *not* test the corner cases. I haven't checked Spark in particular, but we are writing a lot of tests for existing functions and we find lots of such cases.

**Comment by JiajunBernoulli** [comment]
> > > Just a question, how to ensure that the semantics of these functions are consistent with those in SPARK?
> > 
> > 
> > @macroguo-ghy I executed the Function Sqls in the spark engine. And checked the function Definition/OperandType/ReturnType is consistent with the Spark definition.
> 
> Some functions have subtle differences for corner cases of the arguments. And Calcite tests often do _not_ test the corner cases. I haven't checked Spark in particular, but we are writing a lot of tests for existing functions and we find lots of such cases.

In fact, it's difficult for us to handle all situations.
There are also slight differences in functions between different Spark versions.
Many behaviors should be left to users to decide based on their Spark version.

**Comment by mihaibudiu** [comment]
If you use some of the REDUCE optimizations, they will simplify constant expressions involving these functions.

**Comment by YiwenWu** [comment]
> > > > Just a question, how to ensure that the semantics of these functions are consistent with those in SPARK?
> > > 
> > > 
> > > @macroguo-ghy I executed the Function Sqls in the spark engine. And checked the function Definition/OperandType/ReturnType is consistent with the Spark definition.
> > 
> > 
> > Some functions have subtle differences for corner cases of the arguments. And Calcite tests often do _not_ test the corner cases. I haven't checked Spark in particular, but we are writing a lot of tests for existing functions and we find lots of such cases.
> 
> In fact, it's difficult for us to handle all situations. There are also slight differences in functions between different Spark versions. Many behaviors should be left to users to decide based on their Spark version.

yes，Engine version is another question that also confuses us

**Comment by julianhyde** [comment]
It's impossible for Calcite to verify semantics. So, out of scope of this case.

**Comment by YiwenWu** [comment]
> This PR seems good to go. I would like the commit message (and jira summary) to read 'Enable a few...' not 'Add...'.
> 
> And the body of the commit message should list the functions.

done, updated the jira summary and commit message

**Review by macroguo-ghy — APPROVED** [review]
+1

**Pull request merged**